index.js

```diff
@@ -0,0 +1,111 @@
+const http = require("http");
+const open = require("open");
+const fs = require("fs");
+const WebSocketServer = require("websocket").server; // websocket server class
+
+/**
+  CONSTANTS
+**/
+
+// Frontend file paths
+const HTML_FILE_PATH = "index.html";
+const JS_FILE_PATH = "frontend.js";
+
+// Host and default port (if none is specified)
+const DEFAULT_PORT = 9101;
+const HOST = "http://127.0.0.1";
+
+
+/**
+  NETWORKING VARIABLES
+**/
+
+let webSocket = null;
+
+// connection variable, to refer to the web socket connection to the client once connection is established
+let connection = null;
+
+
+// check if a browser window is already open
+let isRunning = false;
+
+/**
+ * @param {String} jsonStr "Stringified JSON data to be viewed"
+ * @param {Number} port "Port which the user wants to use for the network connection between browser and server. Default port of 9101 will be used if not provided by user"
+ */
+exports.visualise = function visualise(jsonStr, port = defaultPort) {
```

Comment:
The _visualise_ function is a little lengthy. I think the following can be placed into their own functions:
1) creation of http server
2) creation of WebSocket server and defining its handlers